### PR TITLE
Re-organize the fedmsg documentation

### PR DESCRIPTION
**Pull request by jeremycline**
This PR has several goals:

 * Organize the documentation so what users want to read is easy to find. This has separated the documentation into three sections (User Guide, API Guide, Contributor Guide). Sections have been ordered in what I think are most-used to least-used.

* Make the landing page short and to the point.

* Remove a lot of the old/outdated documentation

* Merge different documentation sections about the same topic.

* Remove a lot of the Fedora-specific documentation. This can live in the Fedora infra docs project.

It's an unfortunately large pull request and I'm really sorry for that. For the ease of the reviewers, I've got a readthedocs.org build of this branch: http://jcline-fedmsg.readthedocs.io/en/reorg-docs/. You can compare this to https://fedmsg.readthedocs.io/en/latest/ which is the current documentation.

I don't expect this to pass through review without a lot of revision. Please be generous with your feedback 😃.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -1,215 +1,213 @@
...
 
-# The suffix of source filenames.
-source_suffix = '.rst'
-
-# The encoding of source files.
-# source_encoding = 'utf-8'
+# The suffix(es) of source filenames.
+# You can specify multiple suffix as a list of string:
+#
+source_suffix = ['.rst', '.md']
```
Comment (bowlofeggs) [inline doc/conf.py:52]: Whoah I didn't know Sphinx could do markdown. That tempts me to convert my docs to markdown.

File: doc/publishing.rst [hunk 2]
```diff
@@ -1,6 +1,68 @@
...
+.. warning:: Since the network's latency is not 0, it can take some time before
+    a subscriber has set up its connection to the publishing socket. When a
+    publishing socket has no subscribers, it simply drops the published message.
+    Currently, the only way to avoid lost messages is to initialize the socket
+    as early as you can.
 
+To publish a message, call :func:`fedmsg.publish`. Your message should be a
+Python dictionary capable of being JSON-serialized. Additionally, any strings
+it contains should be a text type. That is, :class:`unicode` in Python 2 and
+:class:`str` in Python 3.
```
Comment (bowlofeggs) [inline doc/publishing.rst:20]: Doesn't ```str``` not exist in Python 3?

File: fedmsg/commands/relay.py [hunk 3]
```diff
@@ -32,13 +32,13 @@
 
 
 class RelayCommand(BaseCommand):
-    """ Relay connections from active loggers to the bus.
+    """Relay connections from active loggers to the bus.
```
Comment (bowlofeggs) [inline fedmsg/commands/relay.py:35]: This change seems fine, but unrelated.

File: fedmsg/commands/relay.py [hunk 4]
```diff
@@ -88,6 +88,17 @@ class SigningRelayCommand(RelayCommand):
...
+    """
+    Relay messages from an inbound subscription socket to an outbound publishing socket.
+
+    This service binds to two sockets, :ref:`conf-relay-inbound` and
+    :ref:`conf-relay-outbound`. The inbound socket is a ZeroMQ SUB socket and the
+    outbound socket is a ZeroMQ PUB socket.
+
+    Tools like ``fedmsg-logger`` require that an instance of ``fedmsg-relay`` be running
+    *somewhere* and that it's inbound address be listed in the config as one of the
+    entries in :ref:`conf-relay-inbound`.
```
Comment (bowlofeggs) [inline fedmsg/commands/relay.py:100]: It could be nice to document the type of the return value too.

File: fedmsg/config.py [hunk 5]
```diff
@@ -17,15 +17,16 @@
 #
 # Authors:  Ralph Bean <rbean@redhat.com>
 #
-""" :mod:`fedmsg.config` handles loading, processing and validation of
-all configuration.
+"""
+This module handles loading, processing and validation of all configuration.
```
Comment (bowlofeggs) [inline fedmsg/config.py:21]: Oxford comma [saves capitalists money](http://loweringthebar.net/2017/03/the-oxford-comma-use-it.html).

**Review by bowlofeggs — APPROVED** [review]

File: doc/publishing.rst (hunk 2 above)
Comment (tehsmyers) [reply doc/publishing.rst:20]: It's the other way around (sorta)
```
~ $ python2
>>> str is bytes
True
>>> str is unicode
False
~ $ python3
>>> str is bytes
False
>>> str is unicode
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'unicode' is not defined
>>> 
```

**Pull request merged**